src/backend/executor/nodeSubplan.c

```diff
@@ -965,21 +965,26 @@ ExecSetParamPlan(SubPlanState *node, ExprContext *econtext, QueryDesc *queryDesc
 	SubPlan    *subplan = (SubPlan *) node->xprstate.expr;
 	PlanState  *planstate = node->planstate;
 	SubLinkType subLinkType = subplan->subLinkType;
-	EState	   *estate = planstate->state;
-	ScanDirection dir = estate->es_direction;
+	EState	   *estate;
+	ScanDirection dir;
 	MemoryContext oldcontext = NULL;;
 	TupleTableSlot *slot;
 	ListCell   *l;
 	bool		found = false;
 	ArrayBuildState *astate = NULL;
-	Size		savepeakspace = MemoryContextGetPeakSpace(planstate->state->es_query_cxt);
+	Size		savepeakspace;
 
 	bool		needDtx;
 	bool		shouldDispatch = false;
 	volatile bool explainRecvStats = false;
 
+	Insist(planstate);
+
+	estate = planstate->state;
+	dir = estate->es_direction;
+	savepeakspace = MemoryContextGetPeakSpace(planstate->state->es_query_cxt);
+
 	if (Gp_role == GP_ROLE_DISPATCH &&
-		planstate != NULL &&
```

Comment:
I think you should have just removed this condition.

Reply:
Done

Reply:
Only remove condition without other changes.

Reply:
Why won't we check planstate at least with Assert?

Reply:
Why, if it is definitely not equal to NULL?

Reply:
Can you guarantee that NULL will never be passed as planstate?

Reply:
Yes, it is dereferenced at the very beginning.

Reply:
Yes, that's what I'm talking about. Can you guarantee that it can't be null here?
https://github.com/arenadata/gpdb/blob/c4f634eff2d5f2ee6483facb4b5e38e6c8c97013/src/backend/executor/nodeSubplan.c#L966

Reply:
Done